src/core.c/CompUnit/RepositoryRegistry.pm6

```diff
@@ -148,6 +148,17 @@ class CompUnit::RepositoryRegistry {
              )
            ) -> $home-path {
             $home = $home-path ~ $sep ~ '.raku';
+            # check if we can write the home directory, else fall back to temp directory
+            nqp::mkdir($home, 0o700) unless nqp::stat($home, nqp::const::STAT_EXISTS);
+            if !nqp::stat($home, nqp::const::STAT_ISDIR) {
+                my $tmp = "$*TMPDIR/.raku_{+$*USER}";
```

Comment:
This still suffers from predictable temp directory names. The attacker can know the names of directories and files we are going to create. Thus they will be able to create directories before use and put a symlink in the place of a precomp (or other) file we're going to create. This will cause us to overwrite an arbitrary file chosen by the attacker.

Reply:
Yeah, it probably needs something like https://modules.raku.org/dist/File::Temp:zef:zef:rbt/lib/File/Temp.rakumod#L38